### PR TITLE
HPCC-31195 Add GetConnections, etc to WsDali

### PR DESCRIPTION
**Pull request by wangkx**
Add GetConnections, GetClients, GetSDSStats, and
GetSDSSubscribers to the WsDali service:
1. Add them to ws_dali.ecm, ws_daliservice.cpp, and ws_daliservice.hpp.
2. Include the dadiags.hpp in the ws_daliservice.cpp to use the getDaliDiagnosticValue().

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/scm/ws_dali.ecm [hunk 1]
```diff
@@ -135,8 +135,24 @@ ESPrequest [nil_remove] GetProtectedListRequest
     string CallerId;
 };
 
+ESPrequest [nil_remove]  GetConnectionsRequest
```
Comment (jakesmith) [inline esp/scm/ws_dali.ecm:138]: trivial: double spaces on this line and below seem unusual

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good, aside from trivial formatting issue.
If you agree, can you fix and squash in.

**Comment by wangkx** [comment]
The formatting issue is fixed.
@ghalliday please merge this PR.

**Review by ghalliday — APPROVED** [review]
@wangkx The change looks good.  It strikes me that the try/catch should possibly be in the caller rather than in each of the implementation functions..

**Pull request merged**